Sources/Datadog/FeaturesIntegration/CrashReporting/CrashReportingWithRUMIntegration.swift

```diff
@@ -358,8 +362,12 @@ internal struct CrashReportingWithRUMIntegration: CrashReportingIntegration {
             ),
             context: nil,
             date: startDate.timeIntervalSince1970.toInt64Milliseconds,
-            device: nil, // TODO: RUMM-2197 add device and OS info
-            os: nil, // TODO: RUMM-2197 add device and OS info
+            device: .init(from: deviceInfoProvider, telemetry: nil),
+            // RUMM-2197: In very rare cases, the OS info computed below might not be exactly the one
+            // that the app crashed on. This would correspond to a scenario when the device OS was upgraded
+            // before restarting the app after crash. To solve this, the OS information would have to be
+            // persisted in `crashContext` the same way as we do for other dynamic information.
+            os: .init(from: deviceInfoProvider),
```

Comment:
I haven't decided for this implementation as it will only address a very rare case with quite a work required. It is however a good driver for more robust SDK context management in V2.